back-end/app/repositories/UserRepository.java

```
@@ -0,0 +1,65 @@
+package repositories;
+import models.User;
+import org.hibernate.criterion.Restrictions;
+import play.db.jpa.JPAApi;
+import org.hibernate.Session;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.xml.bind.DatatypeConverter;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+@Singleton
+public class UserRepository {
+    @Inject private JPAApi api;
+
+    public String hashPassword(User user){
+        byte[] emailBytes = user.getEmail().getBytes();
+        byte[] saltBytes = "atlantbh".getBytes();
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] emailOfDigest = md.digest(emailBytes);
+            byte[] saltOfDigest = md.digest(saltBytes);
+            String emailDigested = DatatypeConverter.printHexBinary(emailOfDigest).toLowerCase();
+            String saltDigested= DatatypeConverter.printHexBinary(saltOfDigest).toLowerCase();
+            String finalString=emailDigested+saltDigested+user.getPassword();
+            byte[] finalPasswordBytes = md.digest(finalString.getBytes());
+            String finalpassword = DatatypeConverter.printHexBinary(finalPasswordBytes).toLowerCase();
+            return finalpassword;
+        }
+        catch (NoSuchAlgorithmException e) {
+            System.err.println("I'm sorry, but MD5 is not a valid message digest algorithm");
+            return null;
```

Never hide exceptions. Either print the stack trace or log it using a Logger.